From 40e98184c89c1ff364bd83dee0d1c7f0863bfcec Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Wed, 26 Apr 2023 16:45:25 +0900
Subject: [PATCH 19/20] HACK: Fix build warnings

this patch fixes following warnings;
- -Wunused-but-set-variable
- -Wimplicit-function-declaration

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 examples/cortex-a/armv8/spider/iccom/iccom.c | 2 ++
 os/tpl_os_action.c                           | 2 ++
 os/tpl_os_kernel.c                           | 4 ++++
 os/tpl_os_os_kernel.c                        | 2 ++
 4 files changed, 10 insertions(+)

diff --git a/examples/cortex-a/armv8/spider/iccom/iccom.c b/examples/cortex-a/armv8/spider/iccom/iccom.c
index 2513499c..ccc68c2b 100644
--- a/examples/cortex-a/armv8/spider/iccom/iccom.c
+++ b/examples/cortex-a/armv8/spider/iccom/iccom.c
@@ -29,6 +29,8 @@ enum ICCOM_MSG {
 uint8 iccom_initalized = 0;
 struct echo_command recv_pkt;
 
+void CallTerminateISR2(void);
+
 static void iccom_notify_tx_data(size_t message)
 {
 	iccom_set_out_msg_reg((uint32)message);
diff --git a/os/tpl_os_action.c b/os/tpl_os_action.c
index 5154cf06..3c060acc 100644
--- a/os/tpl_os_action.c
+++ b/os/tpl_os_action.c
@@ -65,6 +65,8 @@ FUNC(void, OS_CODE) tpl_action_activate_task(
 
   /*  init the error to no error  */
   VAR(StatusType, AUTOMATIC) result_action = E_OK;
+  /*  avoid -Wunused-but-set-variable warning */
+  result_action = result_action;
 
   /*  store information for error hook routine    */
   STORE_SERVICE(OSServiceId_ActivateTask)
diff --git a/os/tpl_os_kernel.c b/os/tpl_os_kernel.c
index 8f7cfde0..1c1de365 100644
--- a/os/tpl_os_kernel.c
+++ b/os/tpl_os_kernel.c
@@ -1129,6 +1129,8 @@ FUNC(void, OS_CODE) tpl_init_os(CONST(tpl_application_mode, AUTOMATIC) app_mode)
   CONST(tpl_appmode_mask, AUTOMATIC) app_mode_mask = 1 << app_mode;
 #endif
   VAR(tpl_status, AUTOMATIC) result = E_OK;
+  /*  avoid -Wunused-but-set-variable warning */
+  result = result;
 #if ALARM_COUNT > 0
   P2VAR(tpl_time_obj, AUTOMATIC, OS_APPL_DATA) auto_time_obj;
 #endif
@@ -1291,6 +1293,8 @@ FUNC(void, OS_CODE) tpl_call_terminate_isr2_service(void)
 
   /*  init the error to no error  */
   VAR(StatusType, AUTOMATIC) result = E_OK;
+  /*  avoid -Wunused-but-set-variable warning */
+  result = result;
 
   /*  lock the task structures    */
   LOCK_KERNEL()
diff --git a/os/tpl_os_os_kernel.c b/os/tpl_os_os_kernel.c
index 2746838d..38a5268a 100644
--- a/os/tpl_os_os_kernel.c
+++ b/os/tpl_os_os_kernel.c
@@ -79,6 +79,8 @@ FUNC(tpl_application_mode, OS_CODE) tpl_get_active_application_mode_service(
 {
   VAR(tpl_application_mode, AUTOMATIC) app_mode;
   VAR(tpl_status, AUTOMATIC) result = E_OK;
+  /*  avoid -Wunused-but-set-variable warning */
+  result = result;
 
   /*  lock the kernel    */
   LOCK_KERNEL()
-- 
2.34.1

